python/lsst/obs/base/ingest.py

```diff
@@ -1078,6 +1136,8 @@ def ingestFiles(
             )
 
             try:
+                for name, record in exposure.extraRecords.items():
+                    self.butler.registry.syncDimensionData(name, record, update=update_exposure_records)
```

Comment:
I think I know enough about your use case to guess why you want to insert these dimension records before the exposure records (because your `exposure` has a foreign key referencing one of the extra dimensions, right?).  But I think that's not necessarily the most common extension case.  So, I think we should at least document that this is how  the `extraRecords` field works, and maybe rename it to something like "dependencyRecords", leaving room for a "dependentRecords" field that's inserted only after the `exposure` record is.  That doesn't need to happen on this ticket, but I'd like to leave naming room for it to happen later.

Reply:
You're correct about my use case.
Renaming...